docs/versioning.rst

```diff
@@ -0,0 +1,91 @@
+
+.. _versioning:
+
+Versioning
+==========
+
+µsort tries to maintain the *spirit* of `SemVer <https://semver.org/>`_ while
+focusing primarily on the sorting and formatting results than strictly on the
+API contract.
+
+In general, µsort will attempt to maintain consistent formatting results within
+a major version, with a best-effort attempt to remain "backwards compatible"
+with previous versions of µsort within that major release. Except for obvious
+or egregious formatting errors, any file sorted by a newer release of µsort
+should produce no changes if formatted by previous minor/patch versions
+in that family. Our goal is to enable a safe and consistent rollout of new
```

Comment:
I know what you mean by "rollout" but I think this needs more/different words.  Maybe...

Our goal is to allow central users to codemod for new formatting, before updating the pinned versions for CI.